tests/ExamplesTest.php

```diff
@@ -195,6 +193,259 @@ public function testWithTransaction(): void
 %s
 OUTPUT;
 
-        $this->testExample(__DIR__ . '/../examples/with_transaction.php', $expectedOutput);
+        $this->assertExampleOutput(__DIR__ . '/../examples/with_transaction.php', $expectedOutput);
+    }
+
+    /** @dataProvider provideEncryptionExamples */
+    public function testEncryptionExamples(string $file, string $expectedOutput): void
+    {
+        $this->skipIfClientSideEncryptionIsNotSupported();
+
+        $this->assertExampleOutput($file, $expectedOutput);
+
+        // Clean up metadata and key vault collections
+        $this->dropCollection('test', 'coll', ['encryptedFields' => []]);
+        $this->dropCollection('encryption', '__keyVault');
+    }
+
+    public static function provideEncryptionExamples(): Generator
+    {
+        $expectedOutput = <<<'OUTPUT'
+MongoDB\BSON\Binary Object
```

Comment:
> **Note** this is a good use-case for [snapshot testing](https://github.com/spatie/phpunit-snapshot-assertions).

Reply:
I read through the README but didn't see anything that would apply to these tests since we rely on pattern matching (akin to phpt [EXPECTF](https://qa.php.net/expectf_details.php) codes).

`csfle-explicit_encryption_automatic_decryption.php` is the only one that has entirely static output ("mySecret").

Reply:
Indeed, I hadn't noticed the use of placeholders.